### PR TITLE
29 Semantic data

### LINKED ISSUE #29: Implement semantic datatypes
Implement semantic datatypes for Events and Elements

### PR DESCRIPTION
**Pull request by Exclaminator**
Will close #29

Added semantic data elements.

### CODE REVIEW

File: src/octopeer-github/main/ElementSelectionBehaviour/ButtonsElementSelectionBehaviour.ts [hunk 1]
```diff
@@ -11,7 +11,7 @@
  */
 class ButtonsElementSelectionBehaviour implements ElementSelectionBehaviour {
     private elementDsc: string = ".btn";
-    private elementID: ElementID  = 1;
+    private elementID: ElementID  = new ElementID(1);
```
Comment (mpsijm) [inline src/octopeer-github/main/ElementSelectionBehaviour/ButtonsElementSelectionBehaviour.ts]: While you're at it, please remove the duplicate space before the equals sign.

File: src/octopeer-github/test/DatabaseAdaptable/DatabaseAdapterTest.ts [hunk 2]
```diff
@@ -74,7 +74,7 @@ describe("A DatabaseAdapter", function() {
 
         adapter.setDebug();
 
-        adapter.log(1, 1, new Date(), 100, EMPTY_CALLBACK, EMPTY_CALLBACK);
+        adapter.log(new ElementID(1), new EventID(1), new Date(), 100, EMPTY_CALLBACK, EMPTY_CALLBACK);
```
Comment (mpsijm) [inline src/octopeer-github/test/DatabaseAdaptable/DatabaseAdapterTest.ts]: Please consider instantiating an `ElementID` and an `EventID` before the tests and reference it here (less duplicate code).

**Comment by MathiasMeuleman** [comment]
Code is better like this. Please fix the changes proposed by @mpsijm, then it should be mergable

File: src/octopeer-github/main/EventID.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
+/**
+ * Created by Youri on 08/05/2016.
+ */
+/**
+ * Class to hold elementID's so they can be a semanticaly different datatype.
+ */
+class EventID {
```
Comment (mpsijm) [inline src/octopeer-github/main/EventID.ts]: Please consider moving `EventID` and `ElementID` inside a folder, I think they either belong to `DatabaseAdaptable` or `ElementSelectionBehaviour` (my preference goes to the first).

File: src/octopeer-github/main/ElementID.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
+/**
+ * Created by Youri on 08/05/2016.
+ */
+/**
+ * Class to hold elementID's so they can be a semantically different data type.
+ */
+class ElementID {
```
Comment (mpsijm) [inline src/octopeer-github/main/ElementID.ts]: Please consider moving `EventID` and `ElementID` inside a folder, I think they either belong to `DatabaseAdaptable` or `ElementSelectionBehaviour` (my preference goes to the first).

Comment (Exclaminator) [inline src/octopeer-github/main/ElementID.ts]: There needed for the DatabaseAdaptable, ElementSelectionBehaviour and ElementEventBindingBehaviour. wich are currently all subfolders for these "types", we could possibly move them to a type folder.

Comment (mpsijm) [inline src/octopeer-github/main/ElementID.ts]: I see this is fixed :)

**Comment by mdingjan** [comment]
I've reviewed your changes and have made some minor changes (javadoc, typos, etc.).
Besides, I've used the SAT to check the code; TSLint is happy and so is MaDGe. :)

Ready for merging, in my opinion.

**Comment by mpsijm** [comment]
I'm glad you processed my comments into the code. This can now be merged, nice work! :)

**Pull request merged**